opds2_import.py

```diff
@@ -0,0 +1,830 @@
+            cover_link = self._extract_link(
+                sorted_raw_image_links[1],
+                feed_self_url,
+                default_link_rel=Hyperlink.THUMBNAIL_IMAGE,
+            )
+            image_links.append(cover_link)
+
+        self._logger.debug(
+            u"Finished extracting image links from {0}: {1}".format(
+                encode(publication.images), encode(image_links)
+            )
+        )
+
+        return image_links
+
+    def _extract_links(self, publication, feed_self_url):
+        """Extract a list of LinkData objects from a list of webpub-manifest-parser links.
+
+        :param publication: Publication object
+        :type publication: ast_core.Publication
+
+        :param feed_self_url: Feed's self URL
+        :type feed_self_url: str
+
+        :return: List of links metadata
+        :rtype: List[LinkData]
+        """
+        self._logger.debug(
+            u"Started extracting links from {0}".format(encode(publication.links))
+        )
+
+        links = []
+
+        for link in publication.links:
+            link_metadata = self._extract_link(link, feed_self_url)
+            links.append(link_metadata)
+
+        description_link = self._extract_description_link(publication)
+        if description_link:
+            links.append(description_link)
+
+        image_links = self._extract_image_links(publication, feed_self_url)
+        if image_links:
+            links.extend(image_links)
+
+        self._logger.debug(
+            u"Finished extracting links from {0}: {1}".format(
+                encode(publication.links), encode(links)
+            )
+        )
+
+        return links
+
+    def _extract_media_types_and_drm_scheme_from_link(self, link):
+        """Extract information about content's media type and used DRM schema from the link.
+
+        :param link: Link object
+        :type link: ast_core.Link
+
+        :return: 2-tuple containing information about the content's media type and its DRM schema
+        :rtype: List[Tuple[str, str]]
+        """
+        self._logger.debug(
+            u"Started extracting media types and a DRM scheme from {0}".format(
+                encode(link)
+            )
+        )
+
+        media_types_and_drm_scheme = []
+
+        if link.properties:
+            if (
+                link.properties.availability.state
+                == opds2_ast.OPDS2AvailabilityType.AVAILABLE.value
+            ):
+                for acquisition_object in link.properties.indirect_acquisition:
+                    nested_acquisition_object = acquisition_object
+
+                    while nested_acquisition_object.child:
+                        nested_acquisition_object = first_or_default(
+                            acquisition_object.child
+                        )
+
+                    drm_scheme = DeliveryMechanism.NO_DRM
+
+                    if acquisition_object.type == DeliveryMechanism.ADOBE_DRM:
```

Comment:
Can we avoid this if/else tree by making a set in DeliveryMechanism containing all the known DRM types?

Reply:
Sure, added `KNOWN_DRM_TYPES` to `DeliveryMechanism`